### DISCUSSION TITLE
Age of information or device connectivity status

### DISCUSSION THREAD

**Discussion by eric-murray** [camaraproject/DeviceIdentifier#35] [Ideas]
The current API response gives IMEI and device information, but no indication as to the age of this information or whether the device itself is currently connected to the network.
- Which is more useful, "age of information","device status" or something else?
- In what ways could a mobile network get this information?

Discuss!

**Comment by Kevsy** [comment]
A factor here is porting: I don't know if a number porting away from an operator results in the removal of that IMEI from internal databases/caches. I assume it _should_ but I suspect there are no firm timescales...

**Reply by eric-murray** [reply]
If the IMEI information is retrieved from the PCRF, then that information is only available if the device is currently connected to the network. So if an IMEI is returned, then `ageOfInformation` will be `0`, and `connectivityState` will be `connected`. I'm not aware of any network element that will maintain the historical link between MSISDN and IMEI for devices not connected.

So I think any historical information would need to come from CRM or similar systems. Certainly these systems should be purged if the MSISDN is ported. But I'm not familiar with these types of systems. I think this is an issue for implementation, rather than the API design itself.

**Reply by sachinvodafone** [reply]
Hi @eric-murray , as you stated that IMEI is only returned from PCRF when the device is connected, so both "ageOfInformation" and "connectivityState" may be less relevant in this context. If the IMEI information is only available when the device is actively connected to the network, it implies that the information is inherently up-to-date (ageOfInformation is not significant) and that the device is currently connected (connectivityState is assumed to be "connected").

**Comment by eric-murray** [comment]
Promoted to #47
